addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -711,6 +721,8 @@ MockServer.include({
                 channel_partner_ids: [[3, this.currentPartnerId]],
             },
         ]);
+        const cp = this._getRecords('mail.channel.partner', [['partner_id', '=', this.currentPartnerId], ['channel_id', '=', channel.id]])[0];
+        this._mockUnlink('mail.channel.partner', [[cp.id]]);
```

Comment:
As commented elsewhere this should be done in `_mockMailChannel_inverseChannelPartnerIds` when records are removed. Here there should be no code change.